test/integration/clustertrustbundles/apiserversigner_test.go

```diff
@@ -0,0 +1,222 @@
+*/
+
+package clustertrustbundles
+
+import (
+	"context"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"k8s.io/api/certificates/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructuredscheme"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	certutil "k8s.io/client-go/util/cert"
+	kubeapiservertesting "k8s.io/kubernetes/cmd/kube-apiserver/app/testing"
+	kubecontrollermanagertesting "k8s.io/kubernetes/cmd/kube-controller-manager/app/testing"
+	"k8s.io/kubernetes/test/integration/framework"
+	"k8s.io/kubernetes/test/utils/ktesting"
+	"k8s.io/kubernetes/test/utils/kubeconfig"
+)
+
+func TestClusterTrustBundlesPublisherController(t *testing.T) {
+	// KUBE_APISERVER_SERVE_REMOVED_APIS_FOR_ONE_RELEASE allows for APIs pending removal to not block tests
+	// TODO: Remove this line once certificates v1alpha1 types to be removed in 1.32 are fully removed
+	t.Setenv("KUBE_APISERVER_SERVE_REMOVED_APIS_FOR_ONE_RELEASE", "true")
+	ctx := ktesting.Init(t)
+
+	certBytes := mustMakeCertificate(t, &x509.Certificate{
+		SerialNumber: big.NewInt(0),
+		Subject: pkix.Name{
+			CommonName: "testsigner-kas",
+		},
+		IsCA:                  true,
+		BasicConstraintsValid: true,
+	})
+
+	certPEM := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: certBytes})
+
+	tmpDir := t.TempDir()
+	cacertPath := filepath.Join(tmpDir, "kube-apiserver-serving.crt")
+	if err := certutil.WriteCert(cacertPath, certPEM); err != nil {
+		t.Fatalf("failed to write the CA cert into a file: %v", err)
+	}
+
+	apiServerFlags := []string{
+		"--disable-admission-plugins", "ServiceAccount",
+		"--authorization-mode=RBAC",
+		"--feature-gates", "ClusterTrustBundle=true",
+	}
+	storageConfig := framework.SharedEtcd()
+	server := kubeapiservertesting.StartTestServerOrDie(t, nil, apiServerFlags, storageConfig)
+	defer server.TearDownFn()
+
+	kubeConfigFile := createKubeConfigFileForRestConfig(t, server.ClientConfig)
+
+	kcm := kubecontrollermanagertesting.StartTestServerOrDie(ctx, []string{
+		"--kubeconfig=" + kubeConfigFile,
+		"--controllers=kube-apiserver-serving-clustertrustbundle-publisher", // these are the only controllers needed for this test
+		"--use-service-account-credentials=true",                            // exercise RBAC of kube-apiserver-serving-clustertrustbundle-publisher controller
+		"--leader-elect=false",                                              // KCM leader election calls os.Exit when it ends, so it is easier to just turn it off altogether
+		"--root-ca-file=" + cacertPath,
+		"--feature-gates=ClusterTrustBundle=true",
+	})
+	defer kcm.TearDownFn()
+
+	// setup finished, tests follow
+	clientSet, err := clientset.NewForConfig(server.ClientConfig)
```

Comment:
Also create a CTB for an unrelated signer and assert at the end of the test that it has the same RV.